plugins/cost-insights/src/components/ProductInsightsCard/ProductInsightsCard.tsx

```diff
@@ -14,125 +14,121 @@
  * limitations under the License.
  */
 
-import React, { useCallback, useEffect, useState } from 'react';
-import { InfoCard, useApi } from '@backstage/core';
-import Alert from '@material-ui/lab/Alert';
+import React, {
+  PropsWithChildren,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { InfoCard } from '@backstage/core';
 import { Typography } from '@material-ui/core';
-import { costInsightsApiRef } from '../../api';
-import { ProductInsightsChart } from './ProductInsightsChart';
+import { default as Alert } from '@material-ui/lab/Alert';
 import { PeriodSelect } from '../PeriodSelect';
+import { ProductInsightsChart } from './ProductInsightsChart';
+import { useProductInsightsCardStyles as useStyles } from '../../utils/styles';
+import { DefaultLoadingAction } from '../../utils/loading';
+import { Duration, Entity, Maybe, Product } from '../../types';
 import {
-  useFilters,
   useLastCompleteBillingDate,
-  useLoading,
   useScroll,
+  useLoading,
+  MapLoadingToProps,
 } from '../../hooks';
-import { useProductInsightsCardStyles as useStyles } from '../../utils/styles';
-import { mapFiltersToProps, mapLoadingToProps } from './selector';
-import { Duration, Maybe, Product, Entity } from '../../types';
 import { pluralOf } from '../../utils/grammar';
 
-type ProductInsightsCardProps = {
+type LoadingProps = (isLoading: boolean) => void;
+
+export type ProductInsightsCardProps = {
   product: Product;
+  initialState: {
+    entity: Maybe<Entity>;
+    duration: Duration;
+  };
+  onSelectAsync: (product: Product, duration: Duration) => Promise<Entity>;
 };
 
-export const ProductInsightsCard = ({ product }: ProductInsightsCardProps) => {
-  const client = useApi(costInsightsApiRef);
+const mapLoadingToProps: MapLoadingToProps<LoadingProps> = ({ dispatch }) => (
+  isLoading: boolean,
+) => dispatch({ [DefaultLoadingAction.CostInsightsProducts]: isLoading });
```

Comment:
Should this be product-specific? Or it doesn't matter since we use the same loading indication for everything anyway?

Reply:
It doesn't need to be product-specific anymore.

Since the global spinner locks the UI, it prevents the user from toggling a period on another panel which would generate a new request. Now that product data is fetched before the render, there's only really one state change - when theres at least one product being fetched or not.